File: vowAndConsCalculator.py
'''
calculate the number of vowels and consonants in a string
'''

def finder(chain):
    if isinstance(chain, str):
        charList = []
        # have to make chain case non-sentitive
        for char in chain.lower():
            if char.isalpha():
                charList.append(char)
        # following line could cause issues if other accented consonants as in spanish ñ
        consList ='bcdfghjklmnñpqrsßtvwxz'
        vowels = []; consonants = []
        for letter in charList:
            if letter in consList:
                consonants.append(letter)
            else:
                vowels.append(letter)
        shortAnswer = "%s\n\tVows: %s - Cons: %s" % (chain, len(vowels), len(consonants))
        if len(charList) != len(chain):
            return "%s - Others: %s" % (shortAnswer, (len(chain) - len(charList)))
        return shortAnswer
    return "%s\n\tis not a string input !" % chain

# Here begin test part 

chain1 = 'ßaùâñaó'
chain2 = 'Nabuchodonosor'
chain3 = True
chain4 = 'part_1'

print("%s\t\t\t\tVows: 5 - Cons: 2" % finder(chain1))
print("%s\t\t\t\tVows: 6 - Cons: 8" % finder(chain2))
print("%s\t\t\t\tis not a string input !" % finder(chain3))
print("%s\t\t\tVows: 1 - Cons: 3 - Others: 2" % finder(chain4))

File: vowAndConsCalculator.js
/**
 * calculate the number of vowels and consonants in a string
 *
 */

function finder(chain){
    if(typeof chain === "string"){
        // \u00DF-\u00FF --> from 'ß' to 'ÿ' - Cf. https://en.wikipedia.org/wiki/List_of_Unicode_characters
        chainAlpha = chain.toLowerCase().match(/[a-zA-Z\u00DF-\u00FF]/g);
        // following line could cause issues if other accented consonants as in spanish ñ
        const consList ='bcdfghjklmnñpqrsßtvwxz'
        let vowels = []; let consonants = [];
        for(char of chainAlpha){
            if(consList.includes(char)){
                consonants.push(char)
            }else{
                vowels.push(char)
            }
        }
        let shortAnswer = `'${chain}'\n\tVows: ${vowels.length} - Cons: ${consonants.length}`
        if(chainAlpha.length !== chain.length){
            return shortAnswer + ` - Others: ${chain.length - chainAlpha.length}`
        }
        return shortAnswer
    }
    return `${chain}\n\tis not a string input !`
}

/**
 * Here begin test part 
 * 
 */

var chain1 = 'ßaùâñaó';
var chain2 = 'Nabuchodonosor';
var chain3 = true;
var chain4 = 'part_1';

console.log(`${finder(chain1)}\t\t\t\tVows: 5 - Cons: 2`);
console.log(`${finder(chain2)}\t\t\t\tVows: 6 - Cons: 8`);
console.log(`${finder(chain3)}\t\t\t\tis not a string input !`);
console.log(`${finder(chain4)}\t\t\tVows: 1 - Cons: 3 - Others: 2`);
